packages/composite-checkout/src/lib/payment-methods/paypal.js

```diff
@@ -47,44 +46,22 @@ export function PaypalLabel() {
 	);
 }
 
-export function PaypalSubmitButton( { disabled } ) {
+export function PaypalSubmitButton( { disabled, onClick } ) {
 	const { formStatus } = useFormStatus();
 	const onEvent = useEvents();
-	const {
-		transactionStatus,
-		setTransactionPending,
-		setTransactionRedirecting,
-		setTransactionError,
-	} = useTransactionStatus();
-	const submitTransaction = usePaymentProcessor( 'paypal' );
+	const { transactionStatus } = useTransactionStatus();
 	const [ items ] = useLineItems();
-	const { __ } = useI18n();
 
-	const onClick = () => {
+	const handleButtonPress = () => {
 		onEvent( { type: 'REDIRECT_TRANSACTION_BEGIN', payload: { paymentMethodId: 'paypal' } } );
-		setTransactionPending();
-		submitTransaction( {
+		onClick( 'paypal', {
 			items,
-		} )
-			.then( ( response ) => {
-				if ( ! response ) {
-					setTransactionError(
-						__(
-							'An error occurred while redirecting to PayPal. Please try again or contact support.'
```

Comment:
Do we lose the specificity of this error message? That is, will the user now see a more generic redirect error that does not refer to PayPal? (That's not necessarily a problem.)

Reply:
That's correct. I considered the value of trying to keep the old message, but IMO I don't think it's worth the overhead of doing so right now. 

We'd need to either continue to do this process manually, or pass additional arguments through `onClick`. The main difference is that now you'll see "while redirecting to the payment partner" instead of "while redirecting to PayPal" and I felt that it's a minor difference.